src/openforms/contrib/kvk/migrations/0007_kvkconfig_profile_service_kvkconfig_search_service.py

```diff
@@ -0,0 +1,75 @@
+# Generated by Django 4.2.10 on 2024-03-12 10:10
+
+from django.conf import settings
+from django.core.cache import caches
+import django.db.models.deletion
+from django.db import migrations, models
+
+
+def set_kvk_service(apps, _):
+    """
+    Derive the API root for KVK interaction.
+
+    This used to be configured in one Service. Due to possible gateways we have to go for
+    two separate Services since we need to provide different urls. The API root url can
+    be the same for both operations so in this case we use one Service configuration.
+    """
+    KVKConfig = apps.get_model("kvk", "KVKConfig")
+    config = KVKConfig.objects.first()
+    if config is None:
+        return
+
+    if not (service := config.service) or not (root := service.api_root):
+        return
+
+    if "zoeken" in root:
+        config.search_service = service
+    elif "basisprofielen" in root:
+        config.profile_service = service
+    else:
+        # case where we have defined generic API root url (e.g. https://kvk.nl/test/api/)
+        config.search_service = service
+        config.profile_service = service
```

Comment:
having `zoeken` or `basisprofielen` in root would already break the API clients, since they use the `v2/zoeken` relative paths themselves, so that'd break the existing clients.

Just assigning the service to both fields is enough:

```suggestion
    config.search_service = service
    config.profile_service = service
```